From 732e21ed099999e0049d30743f1a07eec20030ec Mon Sep 17 00:00:00 2001
From: Rahul T R <r-ravikumar@ti.com>
Date: Thu, 19 May 2022 21:00:47 +0530
Subject: [PATCH 6/7] HACK: drm/bridge: cdns-dsi: Remove PM runtime calls

Bridge initialization is failing with PM runtime
enabled. Disable this for now

Signed-off-by: Rahul T R <r-ravikumar@ti.com>
---
 drivers/gpu/drm/bridge/cadence/cdns-dsi-core.c | 25 ++-----------------------
 1 file changed, 2 insertions(+), 23 deletions(-)

diff --git a/drivers/gpu/drm/bridge/cadence/cdns-dsi-core.c b/drivers/gpu/drm/bridge/cadence/cdns-dsi-core.c
index 9a51d68b3ff6..625fdd404093 100644
--- a/drivers/gpu/drm/bridge/cadence/cdns-dsi-core.c
+++ b/drivers/gpu/drm/bridge/cadence/cdns-dsi-core.c
@@ -710,15 +710,11 @@ static void cdns_dsi_bridge_disable(struct drm_bridge *bridge)
 
 	val = readl(dsi->regs + MCTL_MAIN_EN) & ~IF_EN(input->id);
 	writel(val, dsi->regs + MCTL_MAIN_EN);
-	pm_runtime_put(dsi->base.dev);
 }
 
 static void cdns_dsi_bridge_post_disable(struct drm_bridge *bridge)
 {
-	struct cdns_dsi_input *input = bridge_to_cdns_dsi_input(bridge);
-	struct cdns_dsi *dsi = input_to_dsi(input);
-
-	pm_runtime_put(dsi->base.dev);
+	return;
 }
 
 static void cdns_dsi_hs_init(struct cdns_dsi *dsi)
@@ -802,9 +798,6 @@ static void cdns_dsi_bridge_enable(struct drm_bridge *bridge)
 	u32 tmp, reg_wakeup, div;
 	int nlanes;
 
-	if (WARN_ON(pm_runtime_get_sync(dsi->base.dev) < 0))
-		return;
-
 	mode = &bridge->encoder->crtc->state->adjusted_mode;
 	nlanes = output->dev->lanes;
 
@@ -932,9 +925,6 @@ static void cdns_dsi_bridge_pre_enable(struct drm_bridge *bridge)
 	struct cdns_dsi_input *input = bridge_to_cdns_dsi_input(bridge);
 	struct cdns_dsi *dsi = input_to_dsi(input);
 
-	if (WARN_ON(pm_runtime_get_sync(dsi->base.dev) < 0))
-		return;
-
 	cdns_dsi_init_link(dsi);
 	cdns_dsi_hs_init(dsi);
 }
@@ -1055,10 +1045,6 @@ static ssize_t cdns_dsi_transfer(struct mipi_dsi_host *host,
 	struct mipi_dsi_packet packet;
 	int ret, i, tx_len, rx_len;
 
-	ret = pm_runtime_resume_and_get(host->dev);
-	if (ret < 0)
-		return ret;
-
 	cdns_dsi_init_link(dsi);
 
 	ret = mipi_dsi_create_packet(&packet, msg);
@@ -1159,7 +1145,6 @@ static ssize_t cdns_dsi_transfer(struct mipi_dsi_host *host,
 	}
 
 out:
-	pm_runtime_put(host->dev);
 	return ret;
 }
 
@@ -1285,21 +1270,15 @@ static int cdns_dsi_drm_probe(struct platform_device *pdev)
 	if (ret)
 		goto err_disable_pclk;
 
-	pm_runtime_enable(&pdev->dev);
 	dsi->base.dev = &pdev->dev;
 	dsi->base.ops = &cdns_dsi_ops;
 
 	ret = mipi_dsi_host_register(&dsi->base);
-	if (ret)
-		goto err_disable_runtime_pm;
 
 	clk_disable_unprepare(dsi->dsi_p_clk);
 
 	return 0;
 
-err_disable_runtime_pm:
-	pm_runtime_disable(&pdev->dev);
-
 err_disable_pclk:
 	clk_disable_unprepare(dsi->dsi_p_clk);
 
@@ -1311,7 +1290,7 @@ static int cdns_dsi_drm_remove(struct platform_device *pdev)
 	struct cdns_dsi *dsi = platform_get_drvdata(pdev);
 
 	mipi_dsi_host_unregister(&dsi->base);
-	pm_runtime_disable(&pdev->dev);
+	//pm_runtime_disable(&pdev->dev);
 
 	return 0;
 }
-- 
2.30.2

